src/routes/middlewares/authorized-user.middleware.js

```diff
@@ -28,7 +37,40 @@ const checkAuthorizedUser = async (req, res, next) => {
       'Token is invalid, please provide a valid token',
     );
   }
-  return next();
-};
+  try {
+    const decoded = await jwt.verify(token, process.env.SECRET_KEY);
+    req.user = decoded;
+    next();
+  } catch (error) {
+    return errorResponse(res, 400, error.message);
+  }
+}
 
-export default checkAuthorizedUser;
+/**
+ *
+ * @description check if user is authorized
+ * @export
+ * @param {object} req
+ * @param {object} res
+ * @param {void} next
+ * @returns {void}
+ */
+export async function isAuthorized(req, res, next) {
```

Comment:
@emmsdan We already have an existing middleware that performs this same function called `checkAuthorizedUser`. check the middlewares folder inside the route for it.

Reply:
thanks, for the heads up